modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js

```diff
@@ -0,0 +1,266 @@
+'use strict';
+
+var A = AUI();
+
+var createGridField = function(config) {
+	return new Liferay.DDM.Field.Grid(
+		A.merge(
+			{
+				context: {
+					columns: [
+						{
+							label: '1',
+							value: '1'
+						},
+						{
+							label: '2',
+							value: '2'
+						}
+					],
+					name: 'gridField',
+					rows: [
+						{
+							label: 'A',
+							value: 'A'
+						},
+						{
+							label: 'B',
+							value: 'B'
+						}
+					]
+				}
+			},
+			config || {}
+		)
+	).render(document.body);
+};
+
+describe(
+	'Liferay.DDM.Field.Grid',
+	function() {
+		var gridField;
+
+		afterEach(
+			function(done) {
+				gridField.destroy();
+
+				done();
+			}
+		);
+
+		before(
+			function(done) {
+				AUI().use(
+					'liferay-ddm-form-field-grid-template',
+					'liferay-ddm-form-field-grid',
+					function(A) {
+						Liferay.DDM.Renderer.FieldTypes.register(
+							{
+								'javaScriptClass': 'Liferay.DDM.Field.Grid',
+								'name': 'grid',
+								'templateNamespace': 'ddm.grid'
+							}
+						);
+
+						done();
+					}
+				);
+			}
+		);
+
+		describe(
+			'.showErrorMessage()',
+			function() {
+				it(
+					'should exists an error container if the grid field has an error message',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.set('errorMessage', 'error');
+
+						gridField.showErrorMessage();
+
+						var container = gridField.get('container');
+
+						assert.isNotNull(container.one('.help-block'), 'The selectField has an error');
+
+						done();
+					}
+				);
+			}
+		);
+
+		describe(
+			'.setValue()',
+			function() {
+				it(
+					'should return empty value if set an empty object value',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({});
+
+						assert.equal(gridField.getValue().A, undefined);
```

Comment:
I tried to do this way, but the compare between **{}** and **{}** always will be **false**, because two '{}' are different objects.